tests/performance/run_server_performance_test.py

```diff
@@ -0,0 +1,289 @@
+import argparse
+import csv
+import datetime
+import math
+import random
+import subprocess
+import threading
+import time
+from collections import defaultdict
+
+
+VERBOSE = False
+
+
+def return_duration(func):
+    """
+    This decorator makes the applied function return its duration. The original
+    return value gets lost.
+    """
+    def func_wrapper(*args, **kwargs):
+        before = datetime.datetime.now()
+        func(*args, **kwargs)
+        after = datetime.datetime.now()
+        return (after - before).total_seconds()
+
+    return func_wrapper
+
+
+def print_process_output(message, stdout, stderr):
+    global VERBOSE
+
+    if not VERBOSE:
+        return
+
+    print(message)
+    print('-' * 20 + 'stdout' + '-' * 20)
+    print(stdout)
+    print('-' * 20 + 'stderr' + '-' * 20)
+    print(stderr)
+    print('-' * (40 + len('stdout')))
+
+
+def parse_arguments():
```

Comment:
Add a new argument -t <duration test> which specifies the duration of the simulation and measurement. If not specified the test should run indefinitely long until CTRL-C is pressed. When exiting, the script should print the statistics.